### DISCUSSION TITLE
Bundle Startup.js not found (DevTools Console)

### DISCUSSION THREAD

**Discussion by chregi** [pimcore/pimcore#13944] [Q&A / Support] [answered]
I have created a new bundle. This is stored in the "Bundles" directory under "chregi\myBundle". The bundle was created with the PimcoreBundleGeneratorBundle. In the bundles directory resource\public\js the startup.js file was created automatically and in the "Bundles.php" I registered my bundle with "chregiMyBundle::class => ['all' => true]". Now in the DevTools - Console always appears the message that the startup.js file can not be loaded. The problem is that the path to the file is generated incorrectly. Instead of "/bundles/chregi/mybundle/js/pimcore/startup.js" it tries to read the file from "/bundles/chregimybundle/js/pimcore/startup.js". Where did I configure something wrong? Why is the path assembled incorrectly?

**Comment by mattamon** [comment, marked as answer]
Hey @chregi,

which command did you use to create your bundle?
https://github.com/pimcore/bundle-generator/tree/1.0

```bundle:generate``` or ```pimcore:generate:bundle```?

**Reply by chregi** [reply]
I think I used the `pimcore:generate:bundle` command! What does it mean?

**Reply by mattamon** [reply]
Basically if you are using the pimcore command the bundle will get a prefix with pimcore. If you just want to create a symfony bundle you should use the bundle:generate so the prefix won't get added. :)

**Reply by chregi** [reply]
Okay, but what does this have to do with my problem? For me, the path is composed incorrectly. The prefix is a subfolder! What can I change?

**Reply by mattamon** [reply]
Well I assume that because you have used the ```pimcore:bundle:generate``` the path is generated incorrectly hence why it gets generated with the pimcore subfolder.

**Reply by chregi** [reply]
The rest of the bundle runs without problems (PHP). Why are the data found correctly? What can I do to use these subfolders?

**Reply by mattamon** [reply]
Can you check your Bundle.php? 

There should be a getJsPaths method like the following. Is the path correct here?
```php
<?php

namespace TestBundle;

use Pimcore\Extension\Bundle\AbstractPimcoreBundle;

class TestBundle extends AbstractPimcoreBundle
{
    public function getJsPaths()
    {
        return [
            '/bundles/test/js/pimcore/startup.js'
        ];
    }
}
    
```

**Reply by chregi** [reply]
I have adapted this method and changed the path according to the directory. But now I noticed that the correct path is not accessible via browser. Is there any setting so that this URL is not callable in the browser (e.g. route, URL rewrite)?

Bundle Path: "/bundles/chregi/mybundle/Resources/public/js/pimcore/startup.js"

Attempted call: http://dev.pimcore.local/bundles/chregi/mybundle/Resources/public/js/pimcore/startup.js"

**Reply by mattamon** [reply]
The only thing I can think of would be, that it did not get installed correctly. Maybe try to execute `composer install` again or do it manually via 
```bash
bin/console assets:install --symlink --relative
```

Your bundle should me mentioned in the follow output
```bash
 Trying to install assets as relative symbolic links.

 --- ------------------------------------------ ------------------
      Bundle                                     Method / Error
 --- ------------------------------------------ ------------------
  ✔   FOSJsRoutingBundle                         relative symlink
  ✔   PimcoreCoreBundle                          relative symlink
  ✔   PimcoreAdminBundle                         relative symlink
  ✔   PimcoreEcommerceFrameworkBundle            relative symlink
  ✔   PimcoreCustomerManagementFrameworkBundle   relative symlink
  ✔   OutputDataConfigToolkitBundle              relative symlink
  ✔   Web2PrintToolsBundle                       relative symlink
  ✔   PimcoreDataHubBundle                       relative symlink
  ✔   ObjectMergerBundle                         relative symlink
 --- ------------------------------------------ ------------------
 ```

**Reply by mattamon** [reply]
@chregi any luck?

**Reply by chregi** [reply]
I couldn't get it to work with the subdirectory. I created a new bundle without an additional subdirectory and it works.
